src/main/java/org/ow2/proactive/catalog/service/KeyValueMetadataHelper.java

```diff
@@ -34,17 +37,35 @@
 import org.ow2.proactive.catalog.dto.Metadata;
 import org.ow2.proactive.catalog.repository.entity.KeyValueMetadataEntity;
 import org.ow2.proactive.catalog.service.exception.UnprocessableEntityException;
+import org.ow2.proactive.catalog.service.model.GenericInfoBucketData;
 import org.ow2.proactive.catalog.util.parser.CatalogObjectParserFactory;
 import org.ow2.proactive.catalog.util.parser.CatalogObjectParserInterface;
+import org.ow2.proactive.catalog.util.parser.WorkflowParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 
 /**
  * @author ActiveEon Team
  * @since 19/06/2017
  */
+@Component
 public class KeyValueMetadataHelper {
 
-    public static List<KeyValueMetadataEntity> extractKeyValuesFromRaw(String kind, byte[] rawObject) {
+    @SuppressWarnings("FieldCanBeLocal")
+    private static final String GROUP_KEY = "group";
```

Comment:
what if we use uppercase group all over the places? inside the generic information you can put anything you want and then you just need to match the key in the security file